examples/eks/metaflow.tf

```diff
@@ -0,0 +1,64 @@
+resource "random_string" "suffix" {
+  length  = 8
+  special = false
+  upper = false
+}
+
+locals {
+  resource_prefix = "metaflow"
+  resource_suffix = random_string.suffix.result
+  tags = {
+      "managedBy" = "terraform"
+      "application" = "metaflow-eks-example"
+  }
+  cluster_name = "mf-${local.resource_suffix}"
+}
+
+data "aws_availability_zones" "available" {
+}
+
+
+module "metaflow-datastore" {
+  source = "outerbounds/metaflow/aws//modules/datastore"
+  version = "0.3.1"
+
+  resource_prefix = local.resource_prefix
+  resource_suffix = local.resource_suffix
+
+  metadata_service_security_group_id = module.metaflow-metadata-service.metadata_service_security_group_id
+  metaflow_vpc_id                    = module.vpc.vpc_id
+  subnet1_id                         = module.vpc.private_subnets[0]
+  subnet2_id                         = module.vpc.private_subnets[1]
+
+  standard_tags = local.tags
+}
+
+module "metaflow-common" {
+  source = "outerbounds/metaflow/aws//modules/common"
+  version = "0.3.1"
+}
+
+module "metaflow-metadata-service" {
+  # source = "outerbounds/metaflow/aws//modules/metadata-service"
+  # version = "0.3.1"
+  source = "../../modules/metadata-service"
+
+  resource_prefix = local.resource_prefix
+  resource_suffix = local.resource_suffix
+
+  access_list_cidr_blocks           = []
+  api_basic_auth                    = true
+  database_password                 = module.metaflow-datastore.database_password
```

Comment:
I have been gettings errors like because of : 
```
│ Error: Unsupported attribute
│ 
│   on metaflow.tf line 52, in module "metaflow-metadata-service":
│   52:   database_name                    = module.metaflow-datastore.database_name
│     ├────────────────
│     │ module.metaflow-datastore is a object, known only after apply
│ 
│ This object does not have an attribute named "database_name".
```
I think requiring `metaflow-metadata-service`'s `metadata_service_security_group_id` variable in `metaflow-datastore` creates a circular dependency; I can't use dependson in this scenarios as it shouts that there is a circular dependency.